mozilla-services/socorro
# bug 1462749 - add django admin

This sets up the django admin at `siteadmin/` parallel to the admin that we hand crafted from bars of hypo-allergenic bars of soap lightly scented with lavender.

I added Django admin model things for all the Django-model-backed things:

1. Users
2. Groups
3. API Tokens
4. Status Messages

In order to do that, I had to fix a few other things first:

1. the `makesuperuser` command now marks the user as `is_staff` so he/she can access the admin
2. running `make dockersetup` will now create the permissions and groups we need

This is the first step. The next step is to create Django admin versions of all the non-Django-model backed pages:

1. reprocessing
2. supersearch fields
3. site status
4. crash me now
5. analyze model fetches

After that, when we move tables from alembic/sqlalchemy land to Django land, we'll add Django admin views.

To test, do the following:

1. `make dockerbuild`, `make dockersetup`, and then `make dockerupdatedata`
2. make a staff user with `docker-compose run webapp webapp-django/manage.py makesuperuser EMAIL`
3. log into the site and check out `http://localhost:8000/admin/` vs. `http://localhost:8000/siteadmin/`
